text/0317-third-party-dependency-management.md

```diff
@@ -87,16 +97,19 @@ there's nothing we can do to change this situation.
 
 I suggest we take the following actions:
 
-### 1. Perform a full audit of all third-party dependencies CDK uses
+### 1. Perform a full investigation of all third-party dependencies CDK uses
 
 Our goal should be to minimize the number of third-party dependencies we use.
-I suggest we perform an audit of all CDK code,
+I suggest we perform an investigation of all CDK code,
 and when we determine that a given dependency is only used for a small part of the capabilities it offers,
 like a single function,
 consider vendoring in that functionality into our codebase
 (of course, with the proper license and attribution/copyright)
 in cases when we deem that appropriate.
 
+This will be most crucial, but most difficult, to perform in the CDK CLI code,
+and can require large changes that might affect the functionality of the CLI in the extreme case.
```

Comment:
I think it will be impossible - therefore not a realistic recommendation in the current state. Some other approach is required.